package/MDAnalysis/core/universe.py

```diff
@@ -1640,23 +1649,32 @@ def guess_TopologyAttrs(
                     fg =  attr in force_guess
                     try:
                         values = guesser.guess_attr(attr, fg)
-                    except ValueError as e:
+                    except NoDataError as e:
```

Comment:
Also what should happen when you encounter a KeyError? (i.e. the unrecognised topologyattr check you added)

Reply:
Yup for this exact case -- I didn't want to change existing behaviour too much in case downstream packages were relying on it, and this is an existing test for vdwradii that wasn't introduced with guessers. The NoDataError is meant to catch cases where (e.g.) masses can't be guessed since there aren't types, not if the vdwradii is incomplete, so you can turn it off with `error_if_missing`. Same with the KeyError.

Reply:
I'm sorry, I just don't follow what you mean here.

Reply:
Oops. I hallucinated that you linked a test instead of a spot in the code, sorry about that. Basically IIRC there's two tests that test for this ValueError. One is definitely  https://github.com/MDAnalysis/mdanalysis/blob/e776f124c18c41f8990b487e8557d3ad82fe7d1f/testsuite/MDAnalysisTests/core/test_universe.py#L464-L467 and the other one I can't immediately recall. To preserve existing pre-guesser behaviour, we want the ValueError to pass through this check and get caught by those tests.

These passed in the initial guesser PR because that was pre  #4754.

Reply:
Yes, keeping the original behavior makes sense to me (and I confirm it is the original behavior).

Reply:
> The NoDataError is meant to catch cases where (e.g.) masses can't be guessed since there aren't types, not if the vdwradii is incomplete, so you can turn it off with error_if_missing. Same with the KeyError.

I'm not sure I understand what you mean by "same with the KeyError" - i.e. there is nothing here that catches the KeyError as far as I can tell.

Reply:
For the sake of not stalling things more I'm going to go with it (i.e. the KeyError shouldn't happen under normal circumstances), but I would like to know what you meant here @lilyminium - it's unclear if you meant that the KeyError should behave like the NoDataError, in which case this except should include the KeyError too?